### PR TITLE
Malformed credential error

### PR DESCRIPTION
**Pull request by Marton6**
Return a `credential is malformed` error when credential payload is not a valid RSA private key instead of returning an `asn1: trailing data` or similar unhelpful error.

### CODE REVIEW

File: pkg/secrethub/credentials/encoding.go [hunk 1]
```diff
@@ -226,7 +227,7 @@ type rsaPrivateKeyDecoder struct{}
 func (d rsaPrivateKeyDecoder) Decode(payload []byte) (*RSACredential, error) {
 	key, err := x509.ParsePKCS1PrivateKey(payload)
 	if err != nil {
-		return nil, err
+		return nil, ErrMalformedCredential
```
Comment (mackenbach) [inline pkg/secrethub/credentials/encoding.go:230]: Consider making this an `ErrorPref` and wrapping the original error. This will help us quite a lot when debugging a support ticket.

Comment (Marton6) [reply pkg/secrethub/credentials/encoding.go]: Good point.

File: pkg/secrethub/credentials/encoding.go [hunk 2]
```diff
@@ -22,9 +22,10 @@ var (
 	ErrInvalidCredentialHeaderField      = errCredentials.Code("invalid_credential_header_field").ErrorPref("invalid header field: %s")
 	ErrCannotDecodeCredentialHeader      = errCredentials.Code("invalid_credential_header").ErrorPref("cannot decode credential header: %v")
 	ErrUnsupportedCredentialType         = errCredentials.Code("unsupported_credential_type").ErrorPref("unsupported credential type: %s")
-	ErrCannotDecodeCredentialPayload     = errCredentials.Code("invalid_credential_header").ErrorPref("cannot decode credential payload: %v")
+	ErrCannotDecodeCredentialPayload     = errCredentials.Code("invalid_credential_payload").ErrorPref("cannot decode credential payload: %v")
```
Comment (SimonBarendse) [inline pkg/secrethub/credentials/encoding.go:25]: Did you intend to include this change? It seems unrelated to this PR? Do we want to change this error code and why?

Comment (Marton6) [reply pkg/secrethub/credentials/encoding.go]: I think it was a mistake. I changed it back
